File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    answer = []
    dic = {}
    for i in terms:
        i = i.split()
        dic[i[0]] = int(i[-1])
    today = [int(j) for j in today.split(".")]
    num = 0
    
    for i in privacies:
        num += 1
        m = dic[i[-1]]
        i = i[:-2].split(".")
        i = [int(j) for j in i]
        if m >= 12:
            i[0] += m//12
            i[1] += m%12
        else:
            i[1] = i[1] + m
        if i[1] > 12:
            i[0] += i[1]//12
            i[1] = i[1]%12
        print(i)
            
        if today[0] > i[0]:
            answer.append(num)
        elif today[0] < i[0]:
            continue
        else:
            if today[1] > i[1]:
                answer.append(num)
            elif today[1] < i[1]:
                continue
            else:
                if today[2] >= i[2]:
                    answer.append(num)    
        
    return answer

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.java
import java.util.*;

class Solution {
    public int[] solution(String today, String[] terms, String[] privacies) {
        HashMap<String, Integer> term = new HashMap<>();
        for (String t : terms) {
            String[] str = t.split(" ");
            term.put(str[0], Integer.parseInt(str[1]));
        }
        
        String[] todayStr = today.split("\\.");
        int todayYear = Integer.parseInt(todayStr[0]);
        int todayMonth = Integer.parseInt(todayStr[1]);
        int todayDay = Integer.parseInt(todayStr[2]);
        int todayCount = todayYear*12*28+todayMonth*28+todayDay;
        
        ArrayList<Integer> answer = new ArrayList<>();
        for (int i = 0; i < privacies.length; i++) {
            String[] pri = privacies[i].split(" ");
            String[] date = pri[0].split("\\.");
            int year = Integer.parseInt(date[0]);
            int month = Integer.parseInt(date[1]);
            int day = Integer.parseInt(date[2]);
            
            int due = term.get(pri[1]);
            month += due;
            
            int count = year*12*28+month*28+day;
            
            if (todayCount >= count) answer.add(i+1);
        }
        return answer.stream()
	        .mapToInt(Integer::intValue)
    	    .toArray();
    }
}